scripts/bash/run_enceladus.sh

```diff
@@ -386,33 +401,37 @@ SPARK_CONF="--conf spark.logConf=true"
 
 # Dynamic Resource Allocation
 # check DRA safe prerequisites
-if [ "$DRA_ENABLED" = true ] ; then
-    if [ -n "$NUM_EXECUTORS" ]; then
-        echo "WARNING: num-executors should NOT be set when using Dynamic Resource Allocation. DRA is disabled.";
-        DRA_ENABLED=false
-    fi
-    if [ -z "$DRA_MAX_EXECUTORS" ]; then
-        echo "WARNING: maxExecutors should be set for Dynamic Resource Allocation. DRA is disabled"
-        DRA_ENABLED=false
-    fi
+if [ "$DRA_ENABLED" = true ] && [ -z "$DRA_MAX_EXECUTORS" ] ; then
+    echo "WARNING: maxExecutors should be set for Dynamic Resource Allocation. DRA is disabled"
+    DRA_ENABLED=false
 fi
 
 # configure DRA and adaptive execution if enabled
 if [ "$DRA_ENABLED" = true ] ; then
-    echo "Dynamic Resource Allocation enabled"
-    add_spark_conf_cmd "spark.dynamicAllocation.enabled" "true"
-    add_spark_conf_cmd "spark.shuffle.service.enabled" "true"
-    add_spark_conf_cmd "spark.sql.adaptive.enabled" "true"
-    add_spark_conf_cmd "spark.dynamicAllocation.maxExecutors" "${DRA_MAX_EXECUTORS}"
-    if [ -n "$DRA_MIN_EXECUTORS" ]; then
-        add_spark_conf_cmd "spark.dynamicAllocation.minExecutors" "${DRA_MIN_EXECUTORS}"
-    fi
-    if [ -n "$DRA_ALLOCATION_RATIO" ]; then
-        add_spark_conf_cmd "spark.dynamicAllocation.executorAllocationRatio" "${DRA_ALLOCATION_RATIO}"
-    fi
-    if [ -n "$ADAPTIVE_TARGET_POSTSHUFFLE_INPUT_SIZE" ]; then
-        add_spark_conf_cmd "spark.sql.adaptive.shuffle.targetPostShuffleInputSize" "${ADAPTIVE_TARGET_POSTSHUFFLE_INPUT_SIZE}"
-    fi
+  echo "Dynamic Resource Allocation enabled"
+
+  if [ -n "$NUM_EXECUTORS" ]; then
+      echo "WARNING: num-executors should NOT be set when using Dynamic Resource Allocation and will be ignored."
+      echo "Set dra-num-executors if you know what you are doing or disable dra"
+  fi
+
+  echo "Using values from --dra-executor-memory and --dra-executor-cores. --executor-memory and --executor-cores ignored"
+
+  add_spark_conf_cmd "spark.dynamicAllocation.enabled" "true"
+  add_spark_conf_cmd "spark.shuffle.service.enabled" "true"
+  add_spark_conf_cmd "spark.sql.adaptive.enabled" "true"
```

Comment:
I am wondering if all there should be enabled together under one switch. The `shuffle.service` probably should (but are we always certain that the service is set up - as the documentation states?).

The `sql.adaptive` can helpful, but does it really relate with `dynamicAllocation` so that is switched on/off together?

Reply:
So I did not change this. This is how it was the first set up and is currently working.
Maybe @DzMakatun can tell you more about it

Reply:
Good enough for now IMHO. It's a herper script after all. If one wants to run it with all the tweaks, they can do that via spark-submit.

Reply:
**Having adaptive with DRA is a must.** Here is why: if adaptive is disabled (and the code does not partition data explicitly), any shuffle will generate 200 tasks by default, regardless the data size, even for 100 KB of data. As a result DRA will ask for (200 * executorAllocationRatio) executors and  max out to maxExecutors.